examples/temporal_graph_learning/input_encoder.py

```diff
@@ -0,0 +1,101 @@
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+from graphstorm import model as gsmodel
+from graphstorm.utils import get_rank
+
+from model_utils import get_unique_nfields
+from model_utils import get_trainable_params
+from model_utils import to_per_field_nfeats
+from model_utils import concat_over_fields
+
+import dgl
+
+
+class NodeEncoderInputLayer(gsmodel.embed.GSNodeInputLayer):
+    """The temporal input encoder layer for all nodes in a heterogeneous graph.
+
+    The input layer adds a linear layer on nodes with node features and the linear layer projects the node
+    features to a specified dimension.
+    Then, the input layer adds learnable embeddings as temporal encoding on the projected node features.
+
+
+    Parameters
+    ----------
+    g: DistGraph
+        The distributed graph
+    feat_size : dict of int
+        The original feat sizes of each node type
+    embed_size : int
+        The embedding size
+    dropout : float
+        The dropout parameter
+    """
+    def __init__(
+        self,
+        g,
+        out_size,
+        feat_size,
+        dropout=0.0,
+    ):
+        super(NodeEncoderInputLayer, self).__init__(g)
+        self.activation = nn.LeakyReLU()
+        self.dropout = nn.Dropout(dropout)
+        self.out_size = out_size
+        self.feat_size = feat_size
+
+        self.fields = get_unique_nfields(g.etypes)
+        # field_embeds act as temporal encoding
+        self.field_embeds = nn.ParameterDict()
+        for t in self.fields:
+            self.field_embeds[f"{t}_feat"] = get_trainable_params(1, out_size)
```

Comment:
this temporal encoding is basically assign different learnable embeddings for different timestamps? if so, how can the model understand the order in the sequence?

Reply:
I was expecting neural networks can learn this sequence from training data, because such sequential information could be learned from our temporal aggregation strategy:

hidden embeddings at day 1 is only computed on day1's input features, 

hidden embeddings at day 2 is only computed on day1 & day2's input features, ...

Reply:
can you write comments to explain how temporal encoding works in this example code.